addons/web/static/src/views/fields/relational_utils.js

```diff
@@ -573,6 +573,7 @@ export class X2ManyFieldDialog extends Component {
         }); // maybe pass the model directly in props
 
         this.canCreate = !this.record.resId;
+        this.readonly = !this.canCreate && !this.archInfo.activeActions.edit;
```

Comment:
canCreate is badly named imo. I wouldn't use it. I think I'd even remove it (used only once in the template). 
```suggestion
        this.readonly = this.record.resId && !this.archInfo.activeActions.edit;
```